app/helpers/application_helper.rb

```diff
@@ -1,2 +1,7 @@
 module ApplicationHelper
+  def authorize_user
+    if !user_signed_in? || !current_user.admin?
+      raise ActionController::RoutingError.new("Not Found")
+    end
+  end
 end
```

Comment:
I would actually define this method in `application_controller.rb`, rather than `application_helper` - the helper is generally meant for view-specific methods, rather than controller-specific ones. If you define inside application controller, you automatically get it - no including anything! - and if you define more in `application_helper` you don't have to worry about further bloating controllers that are including code they don't need